diff -Nurp a/arch/x86/kernel/cpu/bugs.c b/arch/x86/kernel/cpu/bugs.c
--- a/arch/x86/kernel/cpu/bugs.c	2018-04-29 07:50:07.000000000 +0200
+++ b/arch/x86/kernel/cpu/bugs.c	2018-04-21 20:52:46.597389955 +0200
@@ -308,6 +310,8 @@ ssize_t cpu_show_meltdown(struct device
 		return sprintf(buf, "Not affected\n");
 	if (boot_cpu_has(X86_FEATURE_KAISER))
 		return sprintf(buf, "Mitigation: PTI\n");
+	if (boot_cpu_has(X86_FEATURE_UDEREF))
+		return sprintf(buf, "Mitigation: PaX_UDEREF\n");
 	return sprintf(buf, "Vulnerable\n");
 }
 
