diff -Nurp a/arch/x86/include/asm/uaccess.h b/arch/x86/include/asm/uaccess.h
--- a/arch/x86/include/asm/uaccess.h	2017-11-19 09:22:03.022600865 +0100
+++ b/arch/x86/include/asm/uaccess.h	2017-11-19 09:21:48.946598788 +0100
@@ -75,12 +75,6 @@ static inline bool __chk_range_not_ok(un
 	__chk_range_not_ok((unsigned long __force)(addr), size, limit); \
 })
 
-#ifdef CONFIG_DEBUG_ATOMIC_SLEEP
-# define WARN_ON_IN_IRQ()	WARN_ON_ONCE(!in_task())
-#else
-# define WARN_ON_IN_IRQ()
-#endif
-
 /**
  * access_ok: - Checks if a user space pointer is valid
  * @type: Type of access: %VERIFY_READ or %VERIFY_WRITE.  Note that
@@ -101,11 +95,8 @@ static inline bool __chk_range_not_ok(un
  * checks that the pointer is in the user space range - after calling
  * this function, memory access functions may still return -EFAULT.
  */
-#define access_ok_noprefault(type, addr, size)					\
-({									\
-	WARN_ON_IN_IRQ();						\
-	likely(!__range_not_ok((addr), (size), user_addr_max()));		\
-})
+#define access_ok_noprefault(type, addr, size) \
+	likely(!__range_not_ok((addr), (size), user_addr_max()))
 #define access_ok(type, addr, size) \
 	__access_ok(type, (unsigned long)(addr), (size_t)(size))
 
